File: backend/src/main/java/tom/projects/monkeylog/security/AuthenticatedUser.java
package tom.projects.monkeylog.security;

import tom.projects.monkeylog.model.user.UserOwned;

import java.util.Objects;
import java.util.UUID;

public class AuthenticatedUser {
    public static UUID getId() {
        return UUID.fromString("00000000-0000-0000-0000-000000000000");
    }

    public static boolean isResourceOwner(UserOwned resource) {
        return Objects.equals(resource.getUserId(), getId());
    }

    public static boolean isOwnerOrNull(UserOwned resource) {
        return resource.getUserId() == null || isResourceOwner(resource);
    }
}

File: backendkt/src/main/kotlin/tom/com/monkeylog/security/AuthenticatedUser.kt
package tom.com.monkeylog.security

import tom.com.monkeylog.model.user.UserOwned
import java.util.*

object AuthenticatedUser {
    val id: UUID = UUID.fromString("00000000-0000-0000-0000-000000000000")


    fun isResourceOwner(resource: UserOwned): Boolean {
        return resource.userId == id
    }

    fun isOwnerOrNull(resource: UserOwned): Boolean {
        return resource.userId == null || isResourceOwner(resource)
    }

}
